### DISCUSSION TITLE
What is via in the prompt?

### DISCUSSION THREAD

**Discussion by shinokada** [starship/starship#1857] [Q&A] [answered]
I noticed that there is "via" in my terminal.

What is it?
Is there any way I can remove it?

![image](https://user-images.githubusercontent.com/147320/98317296-93c26e00-201f-11eb-95cd-cab3f544e6c9.png)

**Comment by mickimnet** [comment, marked as answer]
The "via" is part of the standard configuration. There are also some additional other words like "on" and "with" for different prompt modules.

You can remove them all by creating a `starship.toml` and placing it in your `~/.config` or `$XDG_CONFIG_HOME` directory -- see the [configuration documentation](https://starship.rs/config/).

A quick and easy way to just remove the "via", "on" and "with" words is to download the (default) [starship.toml](https://github.com/mickimnet/myth-prompt-themes/blob/main/default/standard/starship.toml) from my repository, place it in the above mentioned directory and remove the words from the file.

**Reply by shinokada** [reply]
Thank you for your reply.

So in your toml file:

```
...
[python]
disabled = false
pyenv_version_name = false
pyenv_prefix = "pyenv "
python_binary = "python"
scan_for_pyfiles = true
format = '[[](bg:yellow fg:#464347)via $symbol$pyenv_prefix$version( \($virtualenv\))[](bg:#464347 fg:yellow)]($style)'
style = "bg:yellow fg:black"
symbol = "🐍 "
...
```

Do I need to remove `via` in front of `$symbol ...`?

**Reply by mickimnet** [reply]
Yes, just remove the `via `. That one is not the default `starship.toml`, it's the Powerline-default I made. 

Please give me an hour or so, I'll upload the right version.

**Reply by mickimnet** [reply]
@shinokada, I just pushed a special branch in my repository for you: [default config, with-out prepositions](https://github.com/mickimnet/myth-prompt-themes/tree/main/default/wo-prepositions) -- I removed all "via", "with" and "on" in the config. Additionally I fixed/updated the above linked standard config.

**Reply by shinokada** [reply]
Thanks.

I tried it but as you can see in the following image, there are 4 lines and some empty spaces.

![image](https://user-images.githubusercontent.com/147320/98415246-434f1d00-20c0-11eb-8366-e057f58ec29e.png)

**Reply by mickimnet** [reply]
My bad 🤦🏻 I forgot to add something. Fixed it. Please re-download and try again.

**Reply by shinokada** [reply]
@mickimnet 
Thanks, it works.

![image](https://user-images.githubusercontent.com/147320/98438784-dbd0b600-212f-11eb-8b2d-ce9331a87994.png)

![image](https://user-images.githubusercontent.com/147320/98438812-0d498180-2130-11eb-8a49-b8a11f621bad.png)

**Reply by mickimnet** [reply]
You're welcome.

**Reply by TajangSec** [reply]
> @shinokada, I just pushed a special branch in my repository for you: [default config, with-out prepositions](https://github.com/mickimnet/myth-prompt-themes/tree/main/default/wo-prepositions) -- I removed all "via", "with" and "on" in the config. Additionally I fixed/updated the above linked standard config.

404 not found
and, I don't want to display this version shown in the image, how can I turn it all off?

**Comment by brianstewart-cl** [comment]
This is helpful, particularly to know about how to disable it, but I was hoping to learn... what is it supposed to mean? Is it something about the current virtual environment? Or the type of package/module the current directory represents?
